core/primitives/src/utils/min_heap.rs

```diff
@@ -20,9 +20,19 @@ impl<T: Ord> MinHeap<T> {
         self.inner.peek().map(|Reverse(t)| t)
     }
 
+    // This sadly leaks an abstraction.
+    pub fn peek_mut(&mut self) -> Option<PeekMut<'_, Reverse<T>>> {
+        self.inner.peek_mut()
+    }
+
     pub fn len(&self) -> usize {
         self.inner.len()
     }
+
+    // This sadly leaks an abstraction.
+    pub fn extend<I: std::iter::IntoIterator<Item = Reverse<T>>>(&mut self, iter: I) {
```

Comment:
```suggestion
    pub fn extend<I: IntoIterator<Item = Reverse<T>>>(&mut self, iter: I) {
```

IntoIterator is in prelude I think